### PR TITLE
Replace use of snap web framework with warp and wai

### PR DESCRIPTION
**Pull request by erikd**
description
-----------

The `snap` framework is poorly maintained and depends on `HsOpenSSL` regardless of whether TLS is used. Furthermore, `HsOpenSSL` will likely break in early 2025 due to some low level APIs being deprecated:

  https://github.com/haskell-cryptography/HsOpenSSL/issues/95

---------

- [x] compiles (`cabal v2-build` or `stack build`)
- [x] tests run successfully (`cabal v2-test` or `stack test`)
- [ ] documentation added
- [ ] link to an issue
- [ ] add milestone (the current sprint)

### CODE REVIEW

**Comment by erikd** [comment]
Both the EKG and the Prometheus endpoints seem to work.

**Comment by erikd** [comment]
With the help of Priyank on Telegram I was able to test that the `gLiveView` script worked as intended with my changes.

Both `USE_EKG=N ./gLiveView.sh -b alpha` and `USE_EKG=Y ./gLiveView.sh -b alpha` worked as expected. I needed the `-b alpha` to work around an issue with the version of bash on my machine.

**Comment by angerman** [comment]
Looks reasonable to me. Thank you Erik!

File: plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs [hunk 1]
```diff
@@ -67,29 +70,28 @@ instance A.ToJSON Number where
...
-    site = route [ ("/metrics/", webhandler ekg) ]
-    webhandler :: EKG.Server -> Snap ()
-    webhandler srv = do
-        samples <- liftIO $ sampleAll $ EKG.serverMetricStore srv
+    settings :: Warp.Settings
+    settings = Warp.setPort port . Warp.setHost host $ Warp.defaultSettings
+    
+    site :: Wai.Application
+    site _request respond = do
+        -- We ignore the request and simple respond with the data.
```
Comment (andreabedini) [inline plugins/backend-ekg/src/Cardano/BM/Backend/Prometheus.lhs:83]: Does this mean any uri will return the same data? (not only `/metrics`)

**Review by andreabedini — APPROVED** [review]
LGTM, I am ok if there is a change in the REST API but it needs to be documented.

**Pull request merged**